lib/src/robotoff_api_client.dart

```diff
@@ -0,0 +1,182 @@
+import 'dart:convert';
+
+import 'package:http/http.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+
+class RobotoffApiClient {
+  RobotoffApiClient._();
+
+  //TODO(x): Add comments for robotoff APIs
+
+  static Future<InsightsResult> getRandomInsights({
+    InsightType? type,
+    OpenFoodFactsCountry? country,
+    String? valueTag,
+    String? serverDomain,
+    QueryType? queryType,
+  }) async {
+    final Map<String, String> parameters = {};
+
+    if (type != null && type.value != null) {
+      parameters['type'] = type.value!;
+    }
+    if (country != null) {
+      parameters['country'] = country.offTag;
+    }
+    if (valueTag != null) {
+      parameters['value_tag'] = valueTag;
+    }
+    if (serverDomain != null) {
+      parameters['server_domain'] = serverDomain;
+    }
+
+    var insightUri = UriHelper.getRobotoffUri(
+      path: 'api/v1/insights/random/',
+      queryType: queryType,
+      queryParameters: parameters,
+    );
+
+    Response response = await HttpHelper().doGetRequest(
+      insightUri,
+      queryType: queryType,
+    );
+    var result = InsightsResult.fromJson(
+      HttpHelper().jsonDecode(utf8.decode(response.bodyBytes)),
+    );
+
+    return result;
+  }
+
+  static Future<InsightsResult> getProductInsights(
+    String barcode, {
+    QueryType? queryType,
+  }) async {
+    var insightsUri = UriHelper.getRobotoffUri(
+      path: 'api/v1/insights/$barcode',
+      queryType: queryType,
+    );
+
+    Response response = await HttpHelper().doGetRequest(
+      insightsUri,
+      queryType: queryType,
+    );
+
+    return InsightsResult.fromJson(
+      HttpHelper().jsonDecode(utf8.decode(response.bodyBytes)),
+    );
+  }
+
+  static Future<RobotoffQuestionResult> getProductQuestions(
+    String barcode,
+    String lang, {
```

Comment:
Should be `OpenFoodFactsLanguage language` instead.

Reply:
Changed the parameter type to `OpenFoodFactsLanguage ` and name to `language `